client/src/lib/bowling.js

```diff
@@ -0,0 +1,71 @@
+class Bowling {
+    constructor() {
+        this.rolls = [];
+        this.score = 0;
+        this.frame = 0;
+        this.rollsIndexPosition = 0
+    };
+
+    getScore() {
+        if(this.rolls.length > 0) {
+            this.calculateScore()
+        }
+        return this.score;
+    }
+
+    roll(score) {
+        if(score < 0 || score > 10) {
+            throw "roll must be between 0 and 10"
+        }
+        this.rolls.push(parseInt(score));
+    }
+
+    isAStrike() {
+        return this.rolls[this.rollsIndexPosition] === 10;
+    }
+
+    isASpare() {
+        return (this.rolls[this.rollsIndexPosition] + this.rolls[this.rollsIndexPosition + 1]) === 10;
+    }
+
+    isLastFrame() {
+        return this.frame === 10;
+    }
+
+    calculateScore() {
+        while (this.rollsIndexPosition < this.rolls.length - 1 && this.frame < 10) {
+            this.calculateRollScore();      
+            // console.log(`frame: ${this.frame} index: ${this.rollsIndexPosition} score: ${this.score}`) 
+        }
+    };
+
+    calculateRollScore() {
```

Comment:
The logic in this method is hard to understand, probably due to the nesting of conditionals. I couldn't work out what rollsIndexPosition was tracking. If you extracted a class `Frame` which stored its rolls, I think this logic would be simpler.